static/js/publisher-pages/pages/Builds/RepoConnected.tsx

```diff
@@ -0,0 +1,314 @@
+                `/api/${snapId}/builds/trigger-build`,
+                {
+                  method: "POST",
+                  headers: {
+                    "Content-Type": "application/json",
+                    "X-CSRFToken": window.CSRF_TOKEN,
+                  },
+                },
+              );
+
+              if (!response.ok) {
+                throw new Error("Unable to trigger new build");
+              }
+
+              refetch();
+            }}
+          >
+            {triggeringBuild ? (
+              <>
+                <i className="p-icon--spinner u-animation--spin"></i>
+                &nbsp;Requesting
+              </>
+            ) : (
+              "Trigger new build"
+            )}
+          </Button>
+        </Col>
+      </Row>
+      {isLoading && <>Waiting for builds...</>}
+      {isFetched && data && (
+        <>
+          <MainTable
+            sortable
+            headers={[
+              { content: "ID", sortKey: "id" },
+              { content: "Architecture", sortKey: "arch" },
+              { content: "Build duration", sortKey: "duration" },
+              {
+                content: "Result",
+                sortKey: "result",
+                className: "p-table__cell--icon-placeholder",
+              },
+              {
+                content: "Build finished",
+                sortKey: "status",
+                className: "u-align--right",
+              },
+            ]}
+            rows={data.snap_builds.map(
+              (build: {
+                id: string;
+                arch_tag: string;
+                duration: string;
+                status: string;
+                datebuilt: string;
+              }) => {
+                return {
+                  columns: [
+                    {
+                      content: (
+                        <Link to={`/${snapId}/builds/${build.id}`}>
+                          {build.id}
+                        </Link>
+                      ),
+                    },
+                    { content: build.arch_tag },
+                    { content: formatDurationString(build.duration) },
+                    {
+                      content: formatStatus(build.status),
+                      className: "p-table__cell--icon-placeholder",
+                    },
+                    {
+                      content: build.datebuilt
+                        ? formatDistanceToNow(build.datebuilt, {
+                            addSuffix: true,
+                          })
+                        : "-",
+                      className: "u-align--right",
+                    },
+                  ],
+                };
+              },
+            )}
+          />
+        </>
+      )}
+      {disconnectModalOpen && (
+        <Modal
+          close={() => {
+            setDisconnectModalOpen(false);
+          }}
+          title="Disconnecting your repository"
+          buttonRow={
+            <>
```

Comment:
Should we create a separate component for button row?

Reply:
Done